src/mesh/index_derivs.cxx

```diff
@@ -649,10 +649,9 @@ const Field2D Mesh::applyXdiff(const Field2D &var, Mesh::deriv_func func,
     return Field2D(0., this);
   }
 
-  CELL_LOC diffloc = var.getLocation();
-
   Field2D result(this);
   result.allocate(); // Make sure data allocated
+  result.setLocation(loc);
```

Comment:
This seems to (potentially) be changing the output location -- it's now certain to be at the requested `loc` which sounds sensible, but by default I think `loc` is `CELL_DEFAULT` and I don't think this makes sense in a call to `setLocation`

Reply:
Good point. Fixed in 0e1adbe, and similar issues with indexVDD*(Field2D) in c334b8b.